### PR TITLE
Distirbuted histograms

### PR DESCRIPTION
**Pull request by Marigold**
Distributed mode for histograms (the single-node still works like before), also fixes some edge cases with NULL values and adds unit & integration tests.

### CODE REVIEW

**Comment by ludovicc** [comment]
@Marigold please rebase from master

**Comment by Marigold** [comment]
Rebased, still need to test it though. Will be done asap.

File: python-histograms/histograms.py [hunk 1]
```diff
@@ -47,7 +52,7 @@ def main():
         histograms_results = compute_histograms(dep_var, indep_vars, nb_bins)
 
         # Store results
-        io_helper.save_results(histograms_results, '', 'application/highcharts+json')
+        io_helper.save_results(histograms_results, '', shapes.Shapes.HIGHCHARTS)
```
Comment (ludovicc) [inline python-histograms/histograms.py]: +100

File: python-histograms/tests/unit/fixtures.py [hunk 2]
```diff
@@ -51,13 +55,20 @@ def inputs_regression(**kwargs):
...
+                    'std': 200.,
+                    'minValue': 700.,
+                    'maxValue': 1300.,
                 }
             ],
             'independent': independent(**kwargs)
         },
         'parameters': []
     }
+    if add_null:
```
Comment (ludovicc) [inline python-histograms/tests/unit/fixtures.py]: refactoring! good, enforce your style

**Comment by ludovicc** [comment]
@Marigold is is ready for release or do you need to add more tests?

**Comment by Marigold** [comment]
Fixed case when `fetch_data` returns `np.nan` instead of `None` for missing values (will add fix to `fetch_data` too so that it is consistent). Tested *non-distributed* case with
```
http -v --verify=no -a admin:WoKeN --timeout 180 POST http://localhost:8087/mining/job \
        user:='{"code":"user1"}' \
        variables:='[{"code":"brainstem"}]' \
        grouping:='[]' \
        covariables:='[{"code":"gender"}]' \
        targetTable='cde_features_a' \
        algorithm:='{"code":"histograms", "name": "Histograms", "parameters": []}'
```
and also tested frontend. Seems to be working fine 👍
